models/models.go

```diff
@@ -114,12 +114,11 @@ func getEngine() (*xorm.Engine, error) {
 		}
 	case "postgres":
 		host, port := "127.0.0.1", "5432"
-		fields := strings.Split(DbCfg.Host, ":")
-		if len(fields) > 0 && len(strings.TrimSpace(fields[0])) > 0 {
-			host = fields[0]
-		}
-		if len(fields) > 1 && len(strings.TrimSpace(fields[1])) > 0 {
-			port = fields[1]
+		if strings.Contains(DbCfg.Host, ":") && !strings.HasSuffix(DbCfg.Host, "]") {
+			host = DbCfg.Host[0:strings.LastIndex(DbCfg.Host, ":")]
```

Comment:
1. `0` is redundant.
2. We should save result of `strings.LastIndex(DbCfg.Host, ":")` to a variable and reuse.